routers/api/v2/tokens.go

```diff
@@ -0,0 +1,64 @@
+package v2
+
+import (
+	"github.com/gin-gonic/gin"
+	v2 "github.com/unicsmcr/hs_auth/authorization/v2"
+	"github.com/unicsmcr/hs_auth/routers/api/models"
+	"go.uber.org/zap"
+	"net/http"
+	"strconv"
+	"strings"
+)
+
+func (r *apiV2Router) ServiceToken(ctx *gin.Context) {
+	owner := ctx.PostForm("owner")
+	allowedURIs := ctx.PostForm("allowedURIs")
+	expiresAt := ctx.PostForm("expiresAt")
+
+	if len(owner) == 0 {
+		r.logger.Debug("service name was not provided in request")
+		models.SendAPIError(ctx, http.StatusBadRequest, "service name for the token must be provided")
+		return
+	}
+
+	if len(allowedURIs) == 0 {
+		r.logger.Debug("no allowedURIs were provided in request")
+		models.SendAPIError(ctx, http.StatusBadRequest, "at least one allowedURI must be provided")
+		return
+	}
+
+	// expiresAt is optional in the form, the default value is -1, i.e. never expires
+	if len(expiresAt) == 0 {
+		expiresAt = "-1"
+	}
+
+	uriList := strings.Split(allowedURIs, ",")
+	var parsedURIs []v2.UniformResourceIdentifier
+	for _, uriString := range uriList {
```

Comment:
You should be able to replace this with `json.Unmarshal([]byte(allowedURIs), parsedURIs)`